src/modules/hip/kernel/flip.hpp

```diff
@@ -0,0 +1,319 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+
+template <typename T>
+__global__ void flip_pkd_tensor(T *srcPtr,
+                                uint2 srcStridesNH,
+                                T *dstPtr,
+                                uint2 dstStridesNH,
+                                uint2 dstDimsWH,
+                                unsigned int *horizontalTensor,
+                                unsigned int *verticalTensor,
+                                RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= dstDimsWH.y) || (id_x >= dstDimsWH.x))
+    {
+        return;
+    }
+
+    d_float24 pix_f24;
+    uint horizontalFlag, verticalFlag, srcIdx;
+    horizontalFlag = horizontalTensor[id_z];
+    verticalFlag = verticalTensor[id_z];
+
+    if(horizontalFlag == 0 && verticalFlag == 0)
```

Comment:
These conditional code can be rewritten more efficiently hip kernel. True for all the following code